perun-core/src/main/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntry.java

```diff
@@ -1150,6 +1152,23 @@ public void removeFacilityContact(PerunSession sess, ContactGroup contactGroupTo
 	}
 
 	@Override
+	public List<Service> getAssignedServices(PerunSession sess, Facility facility) throws InternalErrorException, FacilityNotExistsException, PrivilegeException {
+		Utils.checkPerunSession(sess);
+		getFacilitiesManagerBl().checkFacilityExists(sess, facility);
+		
+		// Authorization
+		if (!AuthzResolver.isAuthorized(sess, Role.VOADMIN, facility)
+				&& !AuthzResolver.isAuthorized(sess, Role.VOOBSERVER, facility)
+				&& !AuthzResolver.isAuthorized(sess, Role.ENGINE)
```

Comment:
Regarding this, we should think, if/when/how can dispatcher/engine use this method instead of getting services from resource. Could it somehow speed up ExecService/Facility resolving from event ?This will be probably up to @mvocu.